### DISCUSSION TITLE
Checking whether Fabric is enabled

### DISCUSSION THREAD

**Discussion by joemun** [software-mansion/react-native-reanimated#5118] [Q&A]
👋  We are in the process of migrating our app to New Architecture, and we've hit a crash where Reanimated thinks it's running in old arch, specifically the code [here](https://github.com/software-mansion/react-native-reanimated/blob/c89032ac29de09509bf0f2b59ea33b0310f0e77a/src/reanimated2/NativeMethods.ts#L166-L182) where the `_IS_FABRIC` check is returning `undefined`.  This code is being run on module load, so I do wonder if there's perhaps a race condition where this value has yet to be set. I do see that it's being set in [RNRuntimeDecorator.cpp](https://github.com/software-mansion/react-native-reanimated/blob/c89032ac29de09509bf0f2b59ea33b0310f0e77a/Common/cpp/ReanimatedRuntime/RNRuntimeDecorator.cpp#L30)

Also, I'm curious what is the proper way of checking for fabric.   RN Screens also checks for [_IS_FABRIC](https://github.com/software-mansion/react-native-screens/blob/53c1c92c47b603418710d6a745306f3647c5285e/src/reanimated/ReanimatedNativeStackScreen.tsx#L27), but RNGH checks for [global.nativeFabricUIManager](https://github.com/software-mansion/react-native-gesture-handler/blob/19455bd1c7d49d7ceee1421170f8b1b4abb3b3c6/FabricExample/src/utils.ts#L8). The RNGH check is interesting because we had to patch on our end to change it from checking `global?.nativeFabricUIManager` to `global._IS_FABRIC`, as it was causing an intermittent crash for some users.

**Comment by tjzel** [comment]
Hi @joemun. We recently had a fix to `_IS_FABRIC` on UI thread https://github.com/software-mansion/react-native-reanimated/pull/5143. Although it doesn't seem to be the case, could you see if it's fixed for you on our nightly version? `yarn add react-native-reanimated@nightly`.

**Reply by joemun** [reply]
@tjzel I'm sorry for the belated response.  We are still hitting this issue.  We just bumped to version `3.6.0-nightly-20231016-02551af17`.

**Reply by tjzel** [reply]
@joemun I'm afraid without a reproduction I can't help you. Injection of e.g. `_IS_FABRIC` to both RN and UI threads is done synchronously so it's hard to have a race condition here. Feel free to open an issue with the reproduction.
